src/main.rs

```diff
@@ -34,9 +34,10 @@ fn main() {
 
     let cfg = match config::Config::new() {
         Ok(c) => c,
+        Err(config::ConfigError::CommandLine(e)) => e.exit(),
         Err(e) => {
-            println!("Failed to read config: {}", e);
-            return;
+            error!("Failed to read config: {}\n", e);
```

Comment:
The logger output isn't very pleasant tough so it might make sense to retain a limited number of prints: for fatal errors and announcing the generator mode and possibly the library name.